src/core/namespace.c

```diff
@@ -1972,6 +2070,16 @@ static int apply_mounts(
                 return log_debug_errno(r, "Failed to open /proc/self/mountinfo: %m");
         }
 
+        /* Prepare a list of verity metadata, that will be used to check the root hash of each image and
+         * drop duplicates. Overlayfs rejects supplying the same directory inode twice as determined by
+         * filesystem UUID and file handle in lowerdir=, even if they are mounted on different paths, as it
+         * resolves each mount to its source filesystem. */
```

Comment:
can you extend the comment and clarify that this stuff here only covers non-DDI images?